src/Parser/EnhancementShaman/Modules/ShamanCore/Pandemic.js

```diff
@@ -0,0 +1,54 @@
+import SPELLS from 'common/SPELLS';
+
+import Module from 'Parser/Core/Module';
+
+const PANDEMIC_THRESHOLD = 11500;
+const debug = false;
+
+class Pandemic extends Module {
+  flametongueTimestamp = 0;
+  earlyRefresh = 0;
+  refreshTimestamps = [];
+
+  on_byPlayer_applybuff(event) {
+    if(event.ability.guid === SPELLS.FLAMETONGUE_BUFF.id) {
+      if(this.flametongueTimestamp !== 0) {
+        if(event.timestamp - this.flametongueTimestamp < PANDEMIC_THRESHOLD) {
```

Comment:
An applybuff event only fires when the buff is applied fresh. So you don't need to worry about pandemics in an applybuff event.

Reply:
True, if the buff fell off completely no point checking if refreshed early.